controllers/datafoundation_client.go

```diff
@@ -0,0 +1,241 @@
+		Owns(&ocsclient.StorageClient{}).
+		Owns(&ocsclient.StorageClassClaim{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+
+	r.storageClient.Name = "ocs-storageclient"
+	r.storageClient.Namespace = offering.Namespace
+
+	r.defaultBlockStorageClassClaim.Name = "default-ocs-block-storageclass-claim"
+	r.defaultBlockStorageClassClaim.Namespace = offering.Namespace
+
+	r.defaultFSStorageClassClaim.Name = "default-ocs-fs-storageclass-claim"
+	r.defaultFSStorageClassClaim.Namespace = offering.Namespace
+}
+
+func (r *dataFoundationClientReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
+	r.Log.Info("Parsing ManagedFusionOffering Data Foundation Client spec")
+
+	isValid := true
+	onboardingTicket, found := offering.Spec.Config["onboardingTicket"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: onboardingTicket"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if onboardingTicket == "" {
+		r.Log.Error(
+			fmt.Errorf("onboardingTicket is empty"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	dfStorageProviderEndpoint, found := offering.Spec.Config["dataFoundationProviderEndpoint"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: dataFoundationProviderEndpoint"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if dfStorageProviderEndpoint == "" {
+		r.Log.Error(
+			fmt.Errorf("dataFoundationProviderEndpoint is empty"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+	dfStorageProviderEndpointAsURL, err := url.ParseRequestURI(dfStorageProviderEndpoint)
+	if err != nil {
+		r.Log.Error(
+			fmt.Errorf("unable to parse Data Foundation Provider Endpoint url"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+	if dfStorageProviderEndpointAsURL.Host == "" {
+		r.Log.Error(
+			fmt.Errorf("cannot retrieve host from %s, might be a malformed or partial URL", dfStorageProviderEndpoint),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
```

Comment:
This check is about the value itself, not the structure. I would suggest moving this check into the reconciliation method that uses the provider endpoint

Reply:
updated